datafusion/expr/src/signature.rs

```diff
@@ -82,31 +82,36 @@ pub enum Volatility {
 /// ```
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub enum TypeSignature {
-    /// arbitrary number of arguments of an common type out of a list of valid types.
```

Comment:
I checked the current functions' signatures, `Variadic*` actually means supports 1 or more arguments instead of "arbitrary number" for current implementations, and use `supports_zero_arguments()` to determine whether 0 arg case is allowed.

So the doc comment is updated for clarification:
`Variadic*` means support 1 or more args
`OneOf(Any(0), Variadic*)` means support 0 or more args